lib/Driver/ToolChains.cpp

```diff
@@ -1218,6 +1218,11 @@ std::string toolchains::GenericUnix::getDefaultLinker() const {
     // final executables, as such, unless specified, we default to gold
     // linker.
     return "gold";
+  case llvm::Triple::x86_64:
+  case llvm::Triple::ppc64:
+  case llvm::Triple::ppc64le:
+    // BFD linker has issues wrt relocations against protected symbols.
+    return "gold";
```

Comment:
In light of what this change is proposing to do, switch to the gold linker by default on Linux, it might be worthwhile to wholly re-write this code.

What I mean is the _default_ should be `"gold"` and then any thing that wants different should opt-out. Something more like:

```
std::string toolchains::GenericUnix::getDefaultLinker() const {
  switch(getTriple().getArch()) {
  default:
    // Otherwise, use the gold linker.
    return "gold";
  }
}
```

Also something that I noticed when changing the patch is that the toolchain is for `GenericUnix` does that mean this changes things for BSD? Is that OK?